framework/src/outputs/Checkpoint.C

```diff
@@ -102,10 +113,23 @@ Checkpoint::outputStep(const ExecFlagType & type)
 bool
 Checkpoint::shouldOutput(const ExecFlagType & type)
 {
+
   // Check if the checkpoint should "normally" output, i.e. if it was created
   // through checkpoint=true
   bool shouldOutput = (onInterval() || type == EXEC_FINAL) ? FileOutput::shouldOutput(type) : false;
 
+  // Check if we should output an autosave based on a time interval.
+  auto curr_time = std::chrono::steady_clock::now();
+  auto elapsed = std::chrono::duration_cast<std::chrono::seconds>(curr_time - start_time);
+  // Print only on timestep end to avoid weird issues
+  if (elapsed >= _autosave_time_interval and type == EXEC_TIMESTEP_END)
+  {
+    start_time = std::chrono::steady_clock::now();
```

Comment:
`_start_time` is a bit of a misnomer - should we call it `_time_since_autosave` or something similar? 

Also, I would argue that we should reset this time each time we checkpoint regardless of how that occurred.  If we are outputting every 5 time steps and the checkpoint is on. We may _never_ end up checkpointing to autosave if those values are coming out frequently enough.

Reply:
I'm not sure if its a misnomer per se, but it may cause confusion since `Output` objects contain like 8 variables already that sound relatively similar. `_time_since_autosave` works for me.

My only concern re: the reset every checkpoint is that we are sure that's the behavior we want before I commit the changes.

Reply:
Yeah, that makes a lot of sense. For a single instance (honestly, you'd be hard pressed to really find a good reason for multiple instances IMO), you really only want to receive outputs frequently enough that you don't have to waste a bunch of CPU time if something goes wrong. For checkpointing, we don't typically care about what time step number these saves come out on. For output purposes, we care a lot!